src/scripts/main.js

```diff
@@ -1,7 +1,307 @@
+            score += board[row][nextCol];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextCol++;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function moveUp() {
+  let moved = false;
+
+  for (let col = 0; col < 4; col++) {
+    for (let row = 1; row < 4; row++) {
+      if (board[row][col] !== 0) {
+        let nextRow = row - 1;
+
+        while (nextRow >= 0) {
+          if (board[nextRow][col] === 0) {
+            board[nextRow][col] = board[row][col];
+            board[row][col] = 0;
+            row = nextRow;
+            moved = true;
+          } else if (board[nextRow][col] === board[row][col]) {
+            board[nextRow][col] *= 2;
+            score += board[nextRow][col];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextRow--;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function moveDown() {
+  let moved = false;
+
+  for (let col = 0; col < 4; col++) {
+    for (let row = 2; row >= 0; row--) {
+      if (board[row][col] !== 0) {
+        let nextRow = row + 1;
+
+        while (nextRow < 4) {
+          if (board[nextRow][col] === 0) {
+            board[nextRow][col] = board[row][col];
+            board[row][col] = 0;
+            row = nextRow;
+            moved = true;
+          } else if (board[nextRow][col] === board[row][col]) {
+            board[nextRow][col] *= 2;
+            score += board[nextRow][col];
+            board[row][col] = 0;
+            moved = true;
+            break;
+          } else {
+            break;
+          }
+          nextRow++;
+        }
+      }
+    }
+  }
+
+  return moved;
+}
+
+function checkWin() {
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 4; c++) {
+      if (board[r][c] === 2048) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function checkGameOver() {
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 4; c++) {
+      if (board[r][c] === 0) {
+        return false;
+      }
+
+      if (
+        (r < 3 && board[r][c] === board[r + 1][c]) ||
+        (c < 3 && board[r][c] === board[r][c + 1])
+      ) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+document.addEventListener("keydown", function (typeOfEvent) {
+  if (typeOfEvent.key === "ArrowUp") {
+    move("up");
+  } else if (typeOfEvent.key === "ArrowDown") {
+    move("down");
+  } else if (typeOfEvent.key === "ArrowLeft") {
+    move("left");
+  } else if (typeOfEvent.key === "ArrowRight") {
+    move("right");
+  }
+});
+
+document.querySelector(".button.start").addEventListener("click", function () {
+  let flag = false;
+
+  for (let r = 0; r < 4; r++) {
+    for (let c = 0; c < 4; c++) {
+      if (board[r][c] !== 0) {
+        flag = true;
+        break;
+      }
+    }
+  }
+
+  if (flag === true) {
```

Comment:
what is `flag`? rename to make it more clear (for example `isGameStarted`, etc.)